src/strategies/call-block-logs/transformer.mjs

```diff
@@ -23,84 +23,41 @@ export function onError(error) {
   throw error;
 }
 
-const contracts = {
-  "0xabefbc9fd2f806065b4f3c237d4b59d9a97bcac7": {
-    name: "zora",
-  },
-  "0x78e3adc0e811e4f93bd9f1f9389b923c9a3355c2": {
-    name: "sound",
-  },
-  "0x0bc2a24ce568dad89691116d5b34deb6c203f342": {
-    name: "catalog",
-    version: "2.0.0",
-  },
-  "0xf5819e27b9bad9f97c177bf007c1f96f26d91ca6": {
-    name: "noizd",
-  },
-  "0x2b5426a5b98a3e366230eba9f95a24f09ae4a584": {
-    name: "mintsongs",
-    version: "2.0.0",
-  },
-};
-
-const events = [
-  "CreatedArtist(uint256,string,string,address)",
-  "Transfer(address,address,uint256)",
-];
-
 /*
 TODO: We need to allow adding new events and contracts dynamically to the
 filtering process. E.g. a platform like sound emits a `CreatedArtist` event and
 from thatpoint on all following events should be checked for a new event too.
 
 */
 
-// NOTE: Logs feature only lowercase addresses
-Object.keys(contracts).forEach((contract) => {
-  if (contract !== contract.toLowerCase()) {
-    throw new Error("Contract address must be lower key");
-  }
-});
-
-function decodeArtistAddress(log) {
-  const topics = log.topics;
-  topics.shift();
-  const result = decodeLog(
-    [
-      {
-        type: "uint256",
-        name: "artistId",
-      },
-      {
-        type: "string",
-        name: "name",
-      },
-      {
-        type: "string",
-        name: "symbol",
-      },
-      {
-        type: "address",
-        name: "artistAddress",
-        indexed: true,
-      },
-    ],
-    log.data,
-    topics
-  );
-  return result.artistAddress.toLowerCase();
-}
+let validatedContracts;
 
 const emptyB32 =
   "0x0000000000000000000000000000000000000000000000000000000000000000";
 
-export function onLine(line) {
+export function onLine(line, contractsPath, events) {
```

Comment:
Ideally, I don't want to use `contractsPath` here but `filter-music-nfts` generates this and we can't simply import that file.